### DISCUSSION TITLE
ImageSharp references System.Numerics.Vectors, version 6.0

### DISCUSSION THREAD

**Discussion by bwcpaluch** [SixLabors/ImageSharp#2800] [Q&A]
Error		
Assembly 'SixLabors.ImageSharp, Version=3.0.0.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13' uses 'System.Numerics.Vectors, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Numerics.Vectors, Version=4.1.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

I don't see that System.Numerics.Vectors has a version 6.0.

I'm using 	SixLabors.ImageSharp.3.1.5 and System.Numerics.Vectors.4.5.0

**Comment by JimBobSquarePants** [comment]
ImageSharp has no direct dependencies other than the runtime itself. 

Why have you installed an old version of System.Numerical.Vectors when it is not required for .NET 6?

**Comment by bwcpaluch** [comment]
But it IS the latest version!  

I can’t find a later one.  It’s not on nuget or anywhere else.

If you know where to find it, please, tell me!

 

B. Paluch

 <tel:718-704-0606> 718 704-0606 Ext. 101

 <http://www.brooklynwebcenter.com/> brooklynwebcenter.com

 

Manage your business online! Give your customers access to their accounts 24/7!

We specialize in custom business solutions for the web - ask for our excellent references!

**Reply by JimBobSquarePants** [reply]
What I mean is you don’t need to install that package. The namespace is already part of the .NET 6 runtime. That package was released out of bounds to provide support for working with vectors on .NET Framework and .NET Standard.

**Comment by bwcpaluch** [comment]
So I can uninstall it, and the problem should fix itself?

 

B. Paluch

 <tel:718-704-0606> 718 704-0606 Ext. 101

 <http://www.brooklynwebcenter.com/> brooklynwebcenter.com

 

Manage your business online! Give your customers access to their accounts 24/7!

We specialize in custom business solutions for the web - ask for our excellent references!

**Comment by bwcpaluch** [comment]
I excluded System.Numerics.Vector and System.Numerics. Tensors from my project, and removed it from my packages.config. Now I get a new error, which I think should be more helpful.


Exception Message: Could not load file or assembly 'SixLabors.ImageSharp, Version=3.1.5.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
Inner Exception: System.IO.FileLoadException: Could not load file or assembly 'SixLabors.ImageSharp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) File name: 'SixLabors.ImageSharp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13' === Pre-bind state information === LOG: DisplayName = SixLabors.ImageSharp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13 (Fully-specified) LOG: Appbase = file:///C:/inetpub/wwwroot/ruachcountry/ruachcountry.org/ LOG: Initial PrivatePath = C:\inetpub\wwwroot\ruachcountry\ruachcountry.org\bin Calling assembly : IronPdf, Version=2024.8.0.3, Culture=neutral, PublicKeyToken=94e1c31412563c75. === LOG: This bind starts in default load context. LOG: Using application configuration file: C:\inetpub\wwwroot\ruachcountry\ruachcountry.org\web.config LOG: Using host configuration file: C:\Users\TEMP\Documents\IISExpress\config\aspnet.config LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config. LOG: Redirect found in application configuration file: 2.0.0.0 redirected to 3.1.5.0. LOG: Post-policy reference: SixLabors.ImageSharp, Version=3.1.5.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13 LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/dbab070e/67a60888/SixLabors.ImageSharp.DLL. LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/dbab070e/67a60888/SixLabors.ImageSharp/SixLabors.ImageSharp.DLL. LOG: Attempting download of new URL file:///C:/inetpub/wwwroot/ruachcountry/ruachcountry.org/bin/SixLabors.ImageSharp.DLL. WRN: Comparing the assembly name resulted in the mismatch: Minor Version ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.
Stack Trace: at IronPdf.Engines.Pdfium.PdfiumInjectionModule.Load() at Ninject.Modules.<7c7b93c9-89be-491a-8a98-63abf8fc010d>NinjectModule.OnLoad(<7c7b93c9-89be-491a-8a98-63abf8fc010d>IKernel kernel) at Ninject.<7c7b93c9-89be-491a-8a98-63abf8fc010d>KernelBase.Load(IEnumerable`1 m) at Ninject.<7c7b93c9-89be-491a-8a98-63abf8fc010d>KernelBase..ctor(<7c7b93c9-89be-491a-8a98-63abf8fc010d>IComponentContainer components, <7c7b93c9-89be-491a-8a98-63abf8fc010d>INinjectSettings settings, <7c7b93c9-89be-491a-8a98-63abf8fc010d>INinjectModule[] modules) at Ninject.<7c7b93c9-89be-491a-8a98-63abf8fc010d>KernelBase..ctor(<7c7b93c9-89be-491a-8a98-63abf8fc010d>INinjectModule[] modules) at Ninject.<7c7b93c9-89be-491a-8a98-63abf8fc010d>StandardKernel..ctor(<7c7b93c9-89be-491a-8a98-63abf8fc010d>INinjectModule[] modules) at IronPdf.Engines.NinjectKernelFactory.<>c.<.cctor>b__7_3() at System.Lazy`1.CreateValue() at System.Lazy`1.LazyInitValue() at IronPdf.Engines.NinjectKernelFactory.get_Kernel() at IronPdf.PdfClientAccessor..ctor() at IronPdf.PdfDocument..ctor(String PdfFilePath, String Password, String OwnerPassword, ChangeTrackingModes TrackChanges) at IronPdf.PdfDocument.FromFile(String PdfFilePath, String Password, String OwnerPassword, ChangeTrackingModes TrackChanges) at ASP.formwrapper_aspx.Page_Load() in c:\inetpub\wwwroot\ruachcountry\ruachcountry.org\FormWrapper.aspx:line 51
Source: IronPdf

 

 

B. Paluch

 <tel:718-704-0606> 718 704-0606 Ext. 101

 <http://www.brooklynwebcenter.com/> brooklynwebcenter.com

 

Manage your business online! Give your customers access to their accounts 24/7!

We specialize in custom business solutions for the web - ask for our excellent references!

**Reply by tiesont** [reply]
You most likely have a binding redirect in one or more *.config files (web.config, for example, since this seems like a .NET Framework project) that needs to be removed or updated.

**Comment by bwcpaluch** [comment]
These are my runtime assembly redirects (the ones that could possibly affect this)

The SixLabors.ImageSharp.   Is the public key correct?

 

<runtime>

   <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">

            <dependentAssembly>

                        <assemblyIdentity name="SixLabors.ImageSharp" publicKeyToken="d998eea7b14cab13" culture="neutral" />

                        <bindingRedirect oldVersion="0.0.0.0-2.0.0.0" newVersion="3.1.5.0" />

            </dependentAssembly>

            

     <dependentAssembly>

       <assemblyIdentity name="IronSoftware.Shared" publicKeyToken="212d70e9efeae4c7" culture="neutral" />

       <bindingRedirect oldVersion="0.0.0.0-2024.1.0.7" newVersion="2024.1.0.7" />

     </dependentAssembly>

      <dependentAssembly>

       <assemblyIdentity name="IronSoftware.Drawing.Common" publicKeyToken="8d7e55c97b3e9835" culture="neutral" />

       <bindingRedirect oldVersion="0.0.0.0-2024.2.0.2" newVersion="2024.8.1.0" />

     </dependentAssembly>

   </assemblyBinding>

</runtime>

 

 

B. Paluch

 <tel:718-704-0606> 718 704-0606 Ext. 101

 <http://www.brooklynwebcenter.com/> brooklynwebcenter.com

 

Manage your business online! Give your customers access to their accounts 24/7!

We specialize in custom business solutions for the web - ask for our excellent references!

**Reply by tiesont** [reply]
The problem is likely this: 

```
<bindingRedirect oldVersion="0.0.0.0-2.0.0.0" newVersion="3.1.5.0" />
```

I'm pretty sure you can (and should) delete the whole `<dependentAssembly>` node for ImageSharp. If for some reason it's required, Visual Studio should offer to create a binding redirect for you that will have all of the correct values. If you don't want to do that, then update the bindingRedirect to

```
<bindingRedirect oldVersion="0.0.0.0-3.1.5.0" newVersion="3.1.5.0" />
```

**Reply by JimBobSquarePants** [reply]
@tiesont  Binding redirects won’t help. If you look at the config and the stack trace the code is targeting .NET framework not .NET 6.

@bwcpaluch ImageSharp v3+ doesn’t work with .NET Framework only .NET 6+. You need to use the latest v2 version. However, if you can, I would strongly recommend targeting the more modern runtime.

**Reply by tiesont** [reply]
@JimBobSquarePants I did notice that (I mentioned as much in a previous comment). I did manage to forget that v3 isn't compatible with .NET Framework, though, so thanks for the reminder.
